src/AutoRest.CSharp/Common/Output/Builders/SerializationBuilder.cs

```diff
@@ -382,7 +384,9 @@ private IEnumerable<JsonPropertySerialization> GetPropertySerializationsFromBag(
                 var serializedName = serializationMapping?.SerializationPath?[^1] ?? schemaProperty.SerializedName;
                 var isRequired = schemaProperty.IsRequired;
                 var shouldExcludeInWireSerialization = (schemaProperty.IsDiscriminator == null || !schemaProperty.IsDiscriminator.Value) && property.InitializationValue is null && schemaProperty.IsReadOnly;
-                var serialization = BuildSerialization(schemaProperty.Schema, property.Declaration.Type, false);
+                var serialization = schemaProperty.Schema.GetType() == typeof(Schema)
+                    ? BuildJsonSerialization(BuilderHelpers.GetInputTypeFromCSharpType(property.Declaration.Type), property.Declaration.Type, false, property.SerializationFormat)
+                    : BuildSerialization(schemaProperty.Schema, property.Declaration.Type, false);
```

Comment:
we should not have this if.
Instead we should construct the property with proper schema when the property is added by customization code.

In the near future, schema related things should be removed, we should not introduce more issues for that work item to resolve.

Reply:
any existing code to generate schema from CSharpType?

Reply:
We need to write one, taking the `GetInputTypeFromCSharpType` as reference.

Reply:
look into the GetInputTypeFromCSharpType, it seems ObjectSchema is used for all type except list and dictionary, is that expected? shouldn't stringSchema used for string and so on?

Reply:
revert the change. we will use schemaProperty == null to do the check